lib/data/dataloader.js

```diff
@@ -0,0 +1,404 @@
+        }
+
+        if (!err && results) {
+            var mbgs = [];
+            var sgvs = [];
+            var cals = [];
+            results.forEach(function(element) {
+                if (element) {
+                    if (element.mbg) {
+                        mbgs.push({
+                            mgdl: Number(element.mbg),
+                            mills: element.date,
+                            device: element.device
+                        });
+                    } else if (element.sgv) {
+                        sgvs.push({
+                            mgdl: Number(element.sgv),
+                            mills: element.date,
+                            device: element.device,
+                            direction: element.direction,
+                            filtered: element.filtered,
+                            unfiltered: element.unfiltered,
+                            noise: element.noise,
+                            rssi: element.rssi
+                        });
+                    } else if (element.type === 'cal') {
+                        cals.push({
+                            mills: element.date,
+                            scale: element.scale,
+                            intercept: element.intercept,
+                            slope: element.slope
+                        });
+                    }
+                }
+            });
+
+            //stop using uniq for SGVs since we use buckets, also enables more detailed monitoring
+            ddata.sgvs = sgvs;
+
+            ddata.mbgs = uniq(mbgs);
+            ddata.cals = uniq(cals);
+        }
+        callback();
+    });
+}
+
+function mergeToTreatments(ddata, results) {
+    var filtered = _.filter(results, function hasId(treatment) {
+        return !_.isEmpty(treatment._id);
+    });
+
+    var treatments = _.map(filtered, function update(treatment) {
+        treatment.mills = new Date(treatment.created_at).getTime();
+        return treatment;
+    });
+
+    //filter out temps older than a day and an hour ago since we don't display them
+    var oldestAgo = ddata.lastUpdated - TWO_DAYS - times.hour().msecs;
+    treatments = _.filter(treatments, function noOldTemps(treatment) {
+        return !treatment.eventType || treatment.eventType.indexOf('Temp Basal') === -1 || treatment.mills > oldestAgo;
+    });
+
+    ddata.treatments = _.unionWith(ddata.treatments, treatments, function(a, b) {
+        return a._id.toString() == b._id.toString();
+    });
+}
+
+
+function loadActivity(ddata, ctx, callback) {
+    var dateRange = {
+        $gte: new Date(ddata.lastUpdated - (ONE_DAY * 2)).toISOString()
+    };
+    if (ddata.page && ddata.page.frame) {
+        dateRange['$lte'] = new Date(ddata.lastUpdated).toISOString();
+    }
+
+    var q = {
+        find: {
+            created_at: dateRange
+        },
+        sort: {
+            created_at: 1
+        }
+    };
+
+    var activity = [];
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['activity' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/manual/m-kozak/cgm-remote-monitor/pullRequest?prid=1196151)